src/LaunchDarkly.ServerSdk.Consul/ConsulDataStoreBuilder.cs

```diff
@@ -0,0 +1,152 @@
+    ///                 Consul.DataStore()
+    ///                     .Address("http://my-consul-host:8500")
+    ///                     .Prefix("app1")
+    ///                 )
+    ///                 .CacheSeconds(15)
+    ///             )
+    ///         .Build();
+    /// </code>
+    /// </remarks>
+    public sealed class ConsulDataStoreBuilder : IPersistentDataStoreAsyncFactory
+    {
+        private ConsulClient _existingClient;
+        private List<Action<ConsulClientConfiguration>> _configActions = new List<Action<ConsulClientConfiguration>>();
+        private Uri _address;
+        private string _prefix = Consul.DefaultPrefix;
+        
+        internal ConsulDataStoreBuilder() { }
+
+        /// <summary>
+        /// Shortcut for calling <see cref="Address(Uri)"/> with a string.
+        /// </summary>
+        /// <param name="address">the URI of the Consul host as a string</param>
+        /// <returns>the builder</returns>
+        /// <seealso cref="Address(Uri)"/>
+        public ConsulDataStoreBuilder Address(string address) => Address(new Uri(address));
+
+        /// <summary>
+        /// Specifies the Consul agent's location.
+        /// </summary>
+        /// <param name="address">the URI of the Consul host</param>
+        /// <returns>the builder</returns>
+        /// /// <seealso cref="Address(string)"/>
+        public ConsulDataStoreBuilder Address(Uri address)
+        {
+            _address = address;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies custom steps for configuring the Consul client. Your action may modify the
+        /// <see cref="ConsulClientConfiguration"/> object in any way.
+        /// </summary>
+        /// <param name="configAction">an action for modifying the configuration</param>
+        /// <returns>the builder</returns>
+        public ConsulDataStoreBuilder ConsulConfigChanges(Action<ConsulClientConfiguration> configAction)
+        {
+            if (configAction != null)
+            {
+                _configActions.Add(configAction);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies an existing, already-configured Consul client instance that the data store
+        /// should use rather than creating one of its own.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// If you specify an existing client, then the other builder methods for configuring Consul are ignored.
+        /// </para>
+        /// <para>
+        /// Note that the LaunchDarkly code will <i>not</i> take ownership of the lifecycle of this
+        /// object: in other words, it will not call <c>Dispose()</c> on the <c>ConsulClient</c> when
+        /// you dispose of the SDK client, as it would if it had created the <c>ConsulClient</c> itself.
+        /// It is your responsibility to call <c>Dispose()</c> on the <c>ConsulClient</c> when you are
+        /// done with it.
```

Comment:
The paragraph above describes new behavior. Previously, the Consul integration would dispose of the Consul client on shutdown regardless of whether it was a pre-existing client instance or not. I've changed it so we only dispose of the client if it was created by us, which I believe is more consistent with typical .NET object lifecycle management.